tests/components/withings/test_init.py

```diff
@@ -158,285 +186,73 @@ async def test_async_setup_no_config(hass: HomeAssistant) -> None:
-                    },
-                ],
-            },
-        },
-        get_sleep_summary_response=WITHINGS_SLEEP_SUMMARY_RESPONSE_EMPTY,
-    )
+    await component_factory.configure_component(profile_configs=(person0,))
 
-    # Test the states of the entities.
-    expected_states = (
-        (profiles[0], const.MEAS_WEIGHT_KG, 70.0),
-        (profiles[0], const.MEAS_FAT_MASS_KG, 5.0),
-        (profiles[0], const.MEAS_FAT_FREE_MASS_KG, 60.0),
-        (profiles[0], const.MEAS_MUSCLE_MASS_KG, 50.0),
-        (profiles[0], const.MEAS_BONE_MASS_KG, 10.0),
-        (profiles[0], const.MEAS_HEIGHT_M, 2.0),
-        (profiles[0], const.MEAS_FAT_RATIO_PCT, 0.07),
-        (profiles[0], const.MEAS_DIASTOLIC_MMHG, 70.0),
-        (profiles[0], const.MEAS_SYSTOLIC_MMGH, 100.0),
-        (profiles[0], const.MEAS_HEART_PULSE_BPM, 60.0),
-        (profiles[0], const.MEAS_SPO2_PCT, 0.95),
-        (profiles[0], const.MEAS_HYDRATION, 0.95),
-        (profiles[0], const.MEAS_PWV, 100.0),
-        (profiles[0], const.MEAS_SLEEP_WAKEUP_DURATION_SECONDS, 320),
-        (profiles[0], const.MEAS_SLEEP_LIGHT_DURATION_SECONDS, 520),
-        (profiles[0], const.MEAS_SLEEP_DEEP_DURATION_SECONDS, 720),
-        (profiles[0], const.MEAS_SLEEP_REM_DURATION_SECONDS, 920),
-        (profiles[0], const.MEAS_SLEEP_WAKEUP_COUNT, 1120),
-        (profiles[0], const.MEAS_SLEEP_TOSLEEP_DURATION_SECONDS, 1320),
-        (profiles[0], const.MEAS_SLEEP_TOWAKEUP_DURATION_SECONDS, 1520),
-        (profiles[0], const.MEAS_SLEEP_HEART_RATE_AVERAGE, 1720),
-        (profiles[0], const.MEAS_SLEEP_HEART_RATE_MIN, 1920),
-        (profiles[0], const.MEAS_SLEEP_HEART_RATE_MAX, 2120),
-        (profiles[0], const.MEAS_SLEEP_RESPIRATORY_RATE_AVERAGE, 2320),
-        (profiles[0], const.MEAS_SLEEP_RESPIRATORY_RATE_MIN, 2520),
-        (profiles[0], const.MEAS_SLEEP_RESPIRATORY_RATE_MAX, 2720),
-        (profiles[1], const.MEAS_HYDRATION, STATE_UNKNOWN),
-        (profiles[3], const.MEAS_FAT_FREE_MASS_KG, STATE_UNKNOWN),
+    config_entry = MockConfigEntry(
+        data={"token": {"userid": "my_user_id"}, "profile": person0.profile}
     )
-    for (profile, meas, value) in expected_states:
-        assert_state_equals(hass, profile, meas, value)
 
-    # Tear down setup entries.
-    entries = hass.config_entries.async_entries(const.DOMAIN)
-    assert entries
-
-    for entry in entries:
-        await hass.config_entries.async_unload(entry.entry_id)
+    with patch("homeassistant.components.withings.async_get_data_manager") as mock:
+        data_manager: DataManager = MagicMock(spec=DataManager)
+        data_manager.poll_data_update_coordinator = MagicMock(
+            spec=DataUpdateCoordinator
+        )
+        data_manager.poll_data_update_coordinator.last_update_success = True
+        mock.return_value = data_manager
+        config_entry.add_to_hass(hass)
 
-    await hass.async_block_till_done()
+        assert await async_setup_entry(hass, config_entry) is True
```

Comment:
Please use the core interface to start the test. Use `hass.config_entries.async_setup`.